lm_eval/tasks/robustness/mmlu_pro/task.py

```diff
@@ -0,0 +1,400 @@
+
+        predict_str = predict_str.strip()
+        to_split = ['the correct answer is', 'the correct response is', 'the correct option is', 'the most accurate answer is',
+                    'the best answer here is', 'the best answer is', 'the answer must be', 'the most accurate response is',
+                    'the most fitting response is', 'the most fitting answer is', 'the answer is', 'answer: ' ]
+        for answer_str in to_split:
+            if answer_str in predict_str.lower():
+                predict_str = predict_str.lower().split(answer_str)
+                predict_str = [i for i in predict_str if len(i) > 0]
+                if len(predict_str) > 0:
+                    predict_str = predict_str[-1].upper()
+                else:
+                    return "EMPTY"
+                predict_str = predict_str.strip(string.punctuation).strip()
+
+        for separator in self.separators:
+            if separator in predict_str:
+                predict_str = predict_str.split(separator)[0].strip()
+
+        delimiters = [" ", ",", "."]
+        quotes = ["'", '"', "'", "`", "`", "."]
+        # if LABEL_TO_ID[task_name] doesn't contain any quotes, remove them from predict_str
+        if not any([quote in "".join(self.labels) for quote in quotes]):
+            for quote in quotes:
+                predict_str = predict_str.replace(quote, "")
+
+        # remove repeated labels while making sure only the label is repeated
+        for label in self.labels:
+            label_count = predict_str.count(label)
+            if label_count > 1:
+                for delimiter in delimiters:
+                    if delimiter in predict_str:
+                        repeated_label = delimiter.join([label] * label_count)
+                        if repeated_label == predict_str:
+                            predict_str = predict_str.split(delimiter)[0]
+                            break
+        replace_dict = {',':'', '\\_': '_', '.':'', ':':''}
+        for key, value in replace_dict.items():
+            predict_str = predict_str.replace(key, value)
+
+        predict_str = predict_str.strip(string.punctuation).strip()
+        return predict_str
+
+    def aggregation(self):
+        """
+        :returns: {str: [float] -> float}
+            A dictionary where keys are the names of submetrics and values are
+            functions that aggregate a list of metrics
+        """
+        aggregations = {}
+        for i in range(10):
+            aggregations[f"accuracy_{i}"] = np.mean
+        return aggregations
+
+    def higher_is_better(self):
+        """
+        :returns: {str: bool}
+            A dictionary where keys are the names of submetrics and values are
+            whether a higher value of the submetric is better
+        """
+        return {
+            "accuracy": True,  # Exact match (the normalized answer exactly match the gold answer
+        }
+
+
+class MmluProSameOption(ConfigurableTask):
```

Comment:
is the difference between this and the task above only swapping labels? if yes, maybe use this as a child class, and all changes above in post processing, metrics etc will be applied here too